src/js/components/invoice/AddItemsPage.jsx

```diff
@@ -127,14 +127,34 @@ const FIELDS = {
         type: TextField,
         label: 'react.invoice.qty.label',
         defaultMessage: 'Qty',
-        flexWidth: '1',
+        flexWidth: '1.1',
         required: true,
         attributes: {
           type: 'number',
           showError: true,
         },
-        getDynamicAttr: ({ rowIndex, values, updateRow }) => ({
-          onBlur: () => updateRow(values, rowIndex),
+        getDynamicAttr: ({
+          rowIndex,
+          values,
+          updateRow,
+          validateInvoiceItem,
+          debouncedInvoiceItemValidation,
+        }) => ({
+          onBlur: (event) => {
+            updateRow(values, rowIndex);
+            validateInvoiceItem({
+              invoiceItemId: values.invoiceItems[rowIndex].id,
+              quantity: event.target.value,
+              rowIndex,
+            });
+          },
+          onChange: (event) => {
+            debouncedInvoiceItemValidation({
+              invoiceItemId: values.invoiceItems[rowIndex].id,
+              quantity: event,
+              rowIndex,
+            });
+          },
```

Comment:
I think the validation onChange is needed here because we don't have any information about validation until we lose focus. Of course, this validation is debounced, so we don't have a lot of requests to send.
Maybe I am wrong, It would be nice to hear opinions from others. @awalkowiak @jmiranda @kchelstowski

Reply:
@alannadolny Considering the fact, that there is a post to the `/api/invoices/validateInvoiceItem` endpoint I am considering requesting a change to do validation during the items / invoice save (and throwing exception if validation won't pass), to avoid adding the rpc endpoints, then you would not have to deal with that part (when to trigger it and on what circumstance)
cc @drodzewicz

Reply:
Now that we went back to validating at the point of data entry, I am not sure what is more appropriate and user-friendly. But I think my main concern would be making sure that a user is not interrupted while they are entering a value here. It might make sense for entering a username for an account signup feature. 

"jmiranda" -> That username is taken
"jmiranda2" -> All good

But when entering a quantity, I would only be ok with a separate validation request on every keypress IFF the validation error was displayed without popups, toast messages, 

"100" -> Too much, my guy
delete a 0 
"10" -> Still not ok, dude
delete another 0 
"1" -> Alright, now we're talking

If this lead to three toast messages appearing on the screen or prompts that required a click to close, I would be annoyed.

However, if there was inline validation that might be less disruptive and ok. 

![image](https://github.com/openboxes/openboxes/assets/1162935/2f2a2cd1-c987-4b84-b268-48b4c79dcf08)